udemy_enroller/runner.py

```diff
@@ -26,18 +32,30 @@ def _redeem_courses(
 
     while True:
         udemy_course_links = loop.run_until_complete(scrapers.run())
-
+        logger.info(f"Total courses this time: {len(udemy_course_links)}")
         if udemy_course_links:
             for course_link in udemy_course_links:
+                counter_total += 1
                 try:
                     status = udemy_actions.enroll(course_link)
+
                     if status == UdemyStatus.ENROLLED.value:
+                        counter_enroled += 1
                         # Try to avoid udemy throttling by sleeping for 1-5 seconds
                         sleep_time = random.choice(range(1, 5))
                         logger.debug(
                             f"Sleeping for {sleep_time} seconds between enrolments"
                         )
                         time.sleep(sleep_time)
+                    elif status == UdemyStatus.ALREADY_ENROLLED.value:
+                        counter_already_enroled += 1
+                    elif status == UdemyStatus.EXPIRED.value:
+                        counter_expired += 1
+                    elif status == UdemyStatus.UNWANTED_LANGUAGE.value:
+                        counter_other_languages += 1
+                    elif status == UdemyStatus.UNWANTED_CATEGORY.value:
+                        counter_other_categories += 1
+
```

Comment:
Could you extract stats logic into an object? Would make it cleaner and more re-usable
Then your class to track the info and the display logic can be inside a method